stdlib/public/SDK/Foundation/NSRange.swift

```diff
@@ -105,18 +105,59 @@ extension NSRange {
 //===----------------------------------------------------------------------===//
 
 extension NSRange {
-    public init(_ x: Range<Int>) {
-        location = x.lowerBound
-        length = x.count
-    }
+  public init<R: RangeExpression>(_ rangeExpression: R)
+  where R.Bound: FixedWidthInteger, R.Bound.Stride : SignedInteger {
+    let range = rangeExpression.relative(to: 0..<R.Bound.max)
+    let start: Int = numericCast(range.lowerBound)
```

Comment:
It would trap – expectedly. To construct an `NSRange` you need an `Int`. The alternative is we only allow construction from `RangeExpression`s of `Int` and push the casting onto the caller. I'm kind of ambivalent which.